tree/ntuple/v7/inc/ROOT/RNTupleDescriptor.hxx

```diff
@@ -372,11 +266,153 @@ public:
    const RPageRange &GetPageRange(ROOT::DescriptorId_t physicalId) const { return fPageRanges.at(physicalId); }
    /// Returns an iterator over pairs { columnId, columnRange }. The iteration order is unspecified.
    RColumnRangeIterable GetColumnRangeIterable() const;
-   bool ContainsColumn(ROOT::DescriptorId_t physicalId) const
+   bool ContainsColumn(ROOT::DescriptorId_t physicalId) const;
+   std::uint64_t GetNBytesOnStorage() const;
+};
+
+// clang-format off
+/**
+\class ROOT::Experimental::RClusterDescriptor::RColumnRange
+\ingroup NTuple
+\brief The window of element indexes of a particular column in a particular cluster
+*/
+// clang-format on
+class RClusterDescriptor::RColumnRange {
+   ROOT::DescriptorId_t fPhysicalColumnId = ROOT::kInvalidDescriptorId;
+   /// The global index of the first column element in the cluster
+   ROOT::NTupleSize_t fFirstElementIndex = ROOT::kInvalidNTupleIndex;
+   /// The number of column elements in the cluster
+   ROOT::NTupleSize_t fNElements = ROOT::kInvalidNTupleIndex;
+   /// The usual format for ROOT compression settings (see Compression.h).
+   /// The pages of a particular column in a particular cluster are all compressed with the same settings.
+   /// If unset, the compression settings are undefined (deferred columns, suppressed columns).
+   std::optional<std::uint32_t> fCompressionSettings;
+   /// Suppressed columns have an empty page range and unknown compression settings.
+   /// Their element index range, however, is aligned with the corresponding column of the
+   /// primary column representation (see Section "Suppressed Columns" in the specification)
+   bool fIsSuppressed = false;
+
+   // TODO(jblomer): we perhaps want to store summary information, such as average, min/max, etc.
+   // Should this be done on the field level?
+
+public:
+   RColumnRange() = default;
+
+   RColumnRange(ROOT::DescriptorId_t physicalColumnId, ROOT::NTupleSize_t firstElementIndex,
+                ROOT::NTupleSize_t nElements, std::optional<std::uint32_t> compressionSettings, bool suppressed = false)
+      : fPhysicalColumnId(physicalColumnId),
+        fFirstElementIndex(firstElementIndex),
+        fNElements(nElements),
+        fCompressionSettings(compressionSettings),
+        fIsSuppressed(suppressed)
    {
-      return fColumnRanges.find(physicalId) != fColumnRanges.end();
    }
-   std::uint64_t GetNBytesOnStorage() const;
+
+   ROOT::DescriptorId_t GetPhysicalColumnId() const { return fPhysicalColumnId; }
+   void SetPhysicalColumnId(ROOT::DescriptorId_t id) { fPhysicalColumnId = id; }
+
+   ROOT::NTupleSize_t GetFirstElementIndex() const { return fFirstElementIndex; }
+   void SetFirstElementIndex(ROOT::NTupleSize_t idx) { fFirstElementIndex = idx; }
+   void IncrFirstElementIndex(ROOT::NTupleSize_t by) { fFirstElementIndex += by; }
```

Comment:
Since this a non-internal API, I would be in favor of spelling it out fully, either `Increment` or `Increase` (same for `IncrNElements`).